src/runtime/runtime.ts

```diff
@@ -402,20 +412,30 @@ export class FunctionConstraint implements Constraint {
     FunctionConstraint.registered[name] = klass;
   }
 
+  static filter = false;
   static variadic = false;
+  static fetchInfo(name:string):typeof FunctionConstraint {
+    return FunctionConstraint.registered[name];
+  }
 
-  static create(name:string, fields:ConstraintFieldMap, restFields:(ID|Register)[] = createArray()):FunctionConstraint|undefined {
+  static create(name:string, fields:ConstraintFieldMap, restFields:(ID|Register)[] = createArray(), packRestIntoFields?:boolean):FunctionConstraint|undefined {
```

Comment:
Originally I thought the fields needed to be correct and available to the constructor, so I was passing the information we needed down the line. That's obviously not true, so I'll move that logic into DSL-land instead.